nano/store/lmdb/lmdb_env.cpp

```diff
@@ -105,5 +106,6 @@ nano::store::write_transaction nano::store::lmdb::env::tx_begin_write (store::lm
 
 MDB_txn * nano::store::lmdb::env::tx (store::transaction const & transaction_a) const
 {
+	debug_assert (transaction_a.store_id () == store_id);
```

Comment:
This looks like the critical check. Would it be enough to use an lmdb::env * as the store_id or does the dispenser give other benefits?

Reply:
The dispenser is used quite broadly on the tracking/logging branch. Naive tracking via pointers to this might work in some cases (like here), but it completely breaks when objects are dynamically allocated.